### PR TITLE
Added function `preprocess_data()` to be imported from `ecog.signal_processing`

### PR DESCRIPTION
**Pull request by luiztauffer**
- Preprocess (resamples, filters) the raw data provided in a NWB file
- Runs spectral decomposition for bands in utils/bands.py
- Stores the processed signals and spectral decomposition power values in the same NWB file

### CODE REVIEW

**Comment by bendichter** [comment]
it looks like cfs and sds are not input args

File: ecog/signal_processing/preprocess_data.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    ...
+
+    Saves to os.path.join(block_path, subject + '_B' + block + '_AA.h5')
+
+    Parameters
+    ----------
+    block_path
+    rate
+    cfs: filter center frequencies. If None, use Chang lab defaults
+    sds: filer standard deviations. If None, use Chang lab defaults
```
Comment (bendichter) [inline ecog/signal_processing/preprocess_data.py]: these values are not in the function def

File: ecog/signal_processing/preprocess_data.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+from pynwb.misc import DecompositionSeries
+
+
+def preprocess_data(path, subject, blocks, phase=False):
+    for block in blocks:
+        block_path = os.path.join(path, '{}_B{}.nwb'.format(subject, block))
+        transform(block_path, phase=phase)
+
+
+def transform(block_path, suffix=None, phase=False, seed=20180928):
```
Comment (bendichter) [inline ecog/signal_processing/preprocess_data.py]: what is suffix?

**Review by bendichter — CHANGES_REQUESTED** [review]
A few minor corrections. Overall it's looking really good!

File: ecog/signal_processing/preprocess_data.py (hunk 1 above)
Comment (luiztauffer) [reply ecog/signal_processing/preprocess_data.py]: Oh I didn't look at the description, will do it now

File: ecog/signal_processing/preprocess_data.py (hunk 2 above)
Comment (luiztauffer) [reply ecog/signal_processing/preprocess_data.py]: it was present in the previous file, it was added at the end of the created file, if the user input something there. I'll remove it

**Comment by JesseLivezey** [comment]
@luiztauffer @bendichter FYI, we're not planning on maintaining anything in this repo. It's likely that in a few weeks we'll make the current version of our preprocessing code public, which was initially based on the code here. Were you looking to build substantially on this code?

**Comment by luiztauffer** [comment]
Hi @JesseLivezey   thanks for the heads up!
We’re altering the current process_ecog for two main reasons:
1 – make the preprocessing function directly usable by an [ECoG visualization](https://github.com/bendichter/ecogVIS) project.
2 – make it store the results in NWB files

Would you mind sharing what will be the main changes with the new software? =)

**Comment by JesseLivezey** [comment]
It might be easier to just copy the preprocessing functions and modify them in your repo. I could also make a different branch and merge them here (I'm using the docker branch for something specific).

The low-level changes to the new version are mostly for operating on arrays that are too large to process in memory all at once. Otherwise, they are the same. The high-level changes are to make the later stages of processing NWB compatible (and also somewhat specific to our lab).

**Comment by JesseLivezey** [comment]
I can't commit to managing anything in this repo, so it might be better for the other project not to depend on anything here.

**Pull request closed without merging**